component/discovery/relabel/relabel.go

```diff
@@ -67,6 +69,7 @@ func (c *Component) Update(args component.Arguments) error {
 
 	targets := make([]discovery.Target, 0, len(newArgs.Targets))
 	relabelConfigs := flow_relabel.ComponentToPromRelabelConfigs(newArgs.RelabelConfigs)
+	c.rcs = relabelConfigs
```

Comment:
It looks like this needs to be protected by a mutex to prevent race conditions on reading/writing `c.rcs` because `getRules` and `Update` can be called in parallel. It's probably the same case in the other components, too.

Reply:
We do this for other components, but hadn't used a mutex in `discovery.relabel` yet. Fixing this.